### PR TITLE
python binding fixes

### PR DESCRIPTION
**Pull request by arvidn**
upgrade travis to trusty and newer xcode. python binding fixes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -41,16 +41,21 @@ cache:
...
-    - libboost1.55-all-dev
-    - libboost1.55-tools-dev
     - python2.7-dev
     - g++-5
 
 before_install:
-
+  # on linux, pull down a recent version of boost and build boost-build
+  - 'if [[ $TRAVIS_OS_NAME == "linux" ]]; then
+    wget https://dl.bintray.com/boostorg/release/1.64.0/source/boost_1_64_0.zip;
```
Comment (aldenml) [inline .travis.yml]: are you willing to increase the minimal boost version? is there something you need in particular?

File: .travis.yml [hunk 2]
```diff
@@ -8,16 +9,15 @@ matrix:
     - env: variant=test_release coverage=1 tests=1 toolset=gcc-coverage python=1
     - env: autotools=1 toolset=gcc
     - os: osx
-      osx_image: xcode7.3
+      osx_image: xcode8.3
```
Comment (aldenml) [inline .travis.yml]: testing with the minimal version of Xcode is useful, unless you want to stay always with the latest version, is that the case?

Comment (arvidn) [inline .travis.yml]: that's a fair point

File: .travis.yml (hunk 1 above)
Comment (arvidn) [inline .travis.yml]: I don't need it yet, but at some point I will need move support in boost::optional. perhaps I shouldn't bump boost version prematurely

**Pull request merged**